dali/python/nvidia/dali/plugin/pytorch.py

```diff
@@ -76,88 +76,100 @@ class DALIGenericIterator(_DaliBaseIterator):
+
+    last_batch_policy = PARTIAL, last_batch_padded = False -> last batch = ``[7]``, next iteration will return ``[2, 3]``
 
-    fill_last_batch = False, last_batch_padded = False -> last batch = ``[7]``, next iteration will return ``[2, 3]``
+    last_batch_policy = FILL, last_batch_padded = True   -> last batch = ``[7, 7]``, next iteration will return ``[1, 2]``
 
-    fill_last_batch = True, last_batch_padded = True   -> last batch = ``[7, 7]``, next iteration will return ``[1, 2]``
+    last_batch_policy = FILL, last_batch_padded = False  -> last batch = ``[7, 1]``, next iteration will return ``[2, 3]``
 
-    fill_last_batch = True, last_batch_padded = False  -> last batch = ``[7, 1]``, next iteration will return ``[2, 3]``
+    last_batch_policy = DROP, last_batch_padded = True   -> last batch = ``[5, 6]``, next iteration will return ``[1, 2]``
+
+    last_batch_policy = DROP, last_batch_padded = False  -> last batch = ``[5, 6]``, next iteration will return ``[2, 3]``
     """
     def __init__(self,
                  pipelines,
                  output_map,
                  size=-1,
                  reader_name=None,
                  auto_reset=False,
-                 fill_last_batch=True,
+                 fill_last_batch=None,
                  dynamic_shape=False,
-                 last_batch_padded=False):
+                 last_batch_padded=False,
+                 last_batch_policy=LastBatchPolicy.FILL):
+
+        # check the assert first as _DaliBaseIterator would run the prefetch
+        assert len(set(output_map)) == len(output_map), "output_map names should be distinct"
+        self._output_categories = set(output_map)
+        self.output_map = output_map
 
-        _DaliBaseIterator.__init__(self, pipelines, size, reader_name, auto_reset, fill_last_batch, last_batch_padded)
+        _DaliBaseIterator.__init__(self, pipelines, size, reader_name, auto_reset, fill_last_batch, last_batch_padded, last_batch_policy)
         self._dynamic_shape = dynamic_shape
 
         # Use double-buffering of data batches
         self._data_batches = [None for i in range(self._num_gpus)]
-        assert len(set(output_map)) == len(output_map), "output_map names should be distinct"
-        self._output_categories = set(output_map)
-        self.output_map = output_map
 
-        # We need data about the batches (like shape information),
-        # so we need to run a single batch as part of setup to get that info
-        for p in self._pipes:
-            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
-                p.schedule_run()
         self._first_batch = None
-        self._first_batch = self.next()
+        try:
+            self._first_batch = self.next()
+        except StopIteration:
+            assert False, "It seems that there is not data in the pipeline, please check if last_batch_policy is set PARTIAL and batch size is bigger than the shard size"
```

Comment:
```suggestion
            assert False, "It seems that there is no data in the pipeline. This may happen if `last_batch_policy` is set to PARTIAL and the requested batch size is greater than the shard size."
```